tests/unit/amo/utils/test_addons.js

```diff
@@ -183,50 +183,157 @@ describe(__filename, () => {
     });
   });
 
-  describe('getPromotedCategory', () => {
-    it('returns null if the addon is not promoted', () => {
+  describe('getPromotedCategories', () => {
+    it('returns the empty list if the addon is not promoted', () => {
       const addon = createInternalAddonWithLang({
         ...fakeAddon,
-        promoted: null,
+        promoted: [],
       });
       const suggestion = createInternalSuggestionWithLang(
-        createFakeAutocompleteResult({ promoted: null }),
+        createFakeAutocompleteResult({ promoted: [] }),
       );
 
       expect(
-        getPromotedCategory({ addon, clientApp: CLIENT_APP_ANDROID }),
-      ).toEqual(null);
+        getPromotedCategories({ addon, clientApp: CLIENT_APP_ANDROID }),
+      ).toEqual([]);
       expect(
-        getPromotedCategory({
+        getPromotedCategories({
           addon: suggestion,
           clientApp: CLIENT_APP_ANDROID,
         }),
-      ).toEqual(null);
+      ).toEqual([]);
     });
 
-    it('returns null if the addon is not promoted for the specified app', () => {
+    it('returns the empty list if the addon is not promoted for the specified app', () => {
       const addon = createInternalAddonWithLang({
         ...fakeAddon,
-        promoted: { category: RECOMMENDED, apps: [CLIENT_APP_ANDROID] },
+        promoted: [{ category: RECOMMENDED, apps: [CLIENT_APP_ANDROID] }],
```

Comment:
Same here, we should keep a test with `promoted` as an object (and not a list)

Reply:
Also in a different test, line 336